File: 2020_November/23.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int ans;
    int dfs(TreeNode* current,int index){
        if (index % 2 == 1) ans += current->val;
        if (current->left != NULL) dfs(current->left,index+1);
        if (current->right != NULL) dfs(current->right,index+1);
    }
    int rob(TreeNode* root) {
        ans = 0;
        dfs(root,1);
        int res = ans;
        dfs(root,0);
        if ()
    }
};

File: 2021_Aug/23.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    
    def findTargetValue(self, root, k):
        if (root == None):
            return None
        
        if (k == root.val):
            return root
        if (k < root.val):
            return self.findTargetValue(root.left, k)
            
        return self.findTargetValue(root.right, k)
        
    
    def findTarget(self, root: Optional[TreeNode], k: int) -> bool:
        stack = []
        stack.append(root)
        
        while (len(stack) > 0):
            head = stack[0]
            stack = stack[1:]
            
            if (head.left != None):
                stack.append(head.left)
            if (head.right != None):
                stack.append(head.right)
            
            tar = self.findTargetValue(root, k - head.val)
            if (tar != None and tar != head):
                return True
            
        return False